### PR TITLE
Setup Hello World and top level testing

### PR DESCRIPTION
**Pull request by parkerl**
I've been taking a stab at this. Getting testing going has been a bit...er...interesting. I will need some guidance setting up Travis. What do you think?

### CODE REVIEW

**Comment by parkerl** [comment]
@kytrinyx had a chance to look at this? Do you think I'm on the right track?

**Comment by kytrinyx** [comment]
Sorry, I just got back from Australia and the time confusion has stolen my brain. I'm way behind on all the exercism stuff. I'll be going through my backlog today.

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,5 @@
 .DS_Store
 bin/configlet
 bin/configlet.exe
+elm-stuff
```
Comment (kytrinyx) [inline .gitignore]: :) Is this something that everyone will have to deal with, or could it go into `.git/info/exclude`?

File: bin/build.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+for example_file in exercises/**/*.example
+do
+  exercise_dir=$(dirname $example_file)
+  exercise=$(basename $example_file .example)
+  mv "$exercise_dir/$exercise.elm" "$exercise_dir/$exercise.impl"
+  mv "$exercise_dir/$exercise.example" "$exercise_dir/$exercise.elm"
+  elm-test exercises/**/*Tests.elm
+  mv "$exercise_dir/$exercise.elm" "$exercise_dir/$exercise.example"
+  mv "$exercise_dir/$exercise.impl" "$exercise_dir/$exercise.elm"
```
Comment (kytrinyx) [inline bin/build.sh]: I love love love the use of `.example` as the file extension, and rename to `.elm`. That's really clean!

File: elm-package.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+    "version": "1.0.0",
+    "summary": "helpful summary of your project, less than 80 characters",
+    "repository": "https://github.com/user/project.git",
```
Comment (kytrinyx) [inline elm-package.json]: https://github.com/exercism/xelm.git

File: elm-package.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+    "version": "1.0.0",
+    "summary": "helpful summary of your project, less than 80 characters",
```
Comment (kytrinyx) [inline elm-package.json]: Exercism exercises in Elm.

**Comment by kytrinyx** [comment]
Yes, this is great! I have a couple of tiny comments, but nothing important.

File: .gitignore (hunk 1 above)
Comment (parkerl) [inline .gitignore]: Oddly this is where Elm puts dependencies and compiler crap. Seems that everyone will have their own. I would think this is the right place.

File: bin/build.sh (hunk 2 above)
Comment (parkerl) [inline bin/build.sh]: Oh thanks, this whole copying around business seems hacky but it looks like the Haskell track is similar. Ugg

**Comment by parkerl** [comment]
Awesome. I will make those updates and then see if we can get Travis off the ground?

File: .gitignore (hunk 1 above)
Comment (kytrinyx) [inline .gitignore]: Huh. Yeah, sounds like we need it. `stuff`. So technical.

File: bin/build.sh (hunk 2 above)
Comment (kytrinyx) [inline bin/build.sh]: And a bunch of other tracks, too.

**Comment by kytrinyx** [comment]
Sounds great.

**Comment by parkerl** [comment]
@kytrinyx Yikes...sorry for the churn...was trying to get Travis going :blush: 

Updates made. Travis build works and is green though it takes over 6 minutes which seems odd. Have you had build times like that with Travis on other projects?

https://travis-ci.org/parkerl/xelm/builds/111125214

**Comment by kytrinyx** [comment]
Yeah, I've had some pretty slow build times. I don't know if it's particular steps of the installation, or just an artifact of running on shared machines.

**Pull request merged**